MiniApp/Classes/MiniAppDownloader.swift

```diff
@@ -56,6 +65,21 @@ class MiniAppDownloader {
             completionHandler(.success(miniAppPath))
         }
     }
+
+    private func cleanVersions(for appId: String, differentFrom versionId: String) {
+        let miniAppStoragePath = FileManager.getMiniAppDirectory(with: appId)
+
+        if let directoryContents = try? FileManager.default.contentsOfDirectory(at: miniAppStoragePath, includingPropertiesForKeys: nil, options: .skipsSubdirectoryDescendants) {
+            for path in directoryContents where path.lastPathComponent != versionId {
+                do {
+                    try FileManager.default.removeItem(at: path)
+                    miniAppStatus.setDownloadStatus(false, appId: appId, versionId: versionId)
```

Comment:
Guess, we need to update the MiniAppStatus class to remove the value when deleting a specific version. I will update the MiniAppStatus class.

Reply:
I temporarily managed a way to manage it in MiniAppStorage, see my next PR